### ISSUE TITLE
aws-cloudfront-s3: Adding custom lambda@edge to default behavior removes insertHttpSecurityHeaders lambda@edge

### ISSUE THREAD

**Issue by andrewgadziksonos** [awslabs/aws-solutions-constructs#114]
Extending the `defaultBehavior` props with a custom lambda@edge removes the `insertHttpSecurityHeaders` lambda@edge configuration from the default cache behavior.

### Reproduction Steps

```js
const handler = new Function(scope, 'SomeHandler', {
  functionName: 'SomeHandler',
  runtime: Runtime.NODEJS_12_X,
  handler: 'index.handler',
  code: Code.fromAsset(path.join(__dirname, '../lambdas/function')),
})

const handlerVersion = new Version(scope, 'SomeHandlerVersion', {
  lambda: handler,
})

new CloudFrontToS3(scope, 'CloudFrontToS3', {
  cloudFrontDistributionProps: {
    defaultBehavior: {
       edgeLambdas: [
         {
           eventType: LambdaEdgeEventType.VIEWER_REQUEST,
           includeBody: false,
           functionVersion: handlerVersion,
         }
       ]
    }
})
```

### Error Log

No errors produced, but CFN does not contain configuration for the `insertHttpSecurityHeaders` lambda@edge

### Environment

  - **CDK CLI Version      :*1.80.0 (build 31132ca)*
  - **CDK Framework Version:*1.77.0*
  - **AWS Solutions Constructs Version     :*1.77.0*
  - **OS                   :*MacOS 10.15.7 (19H114)*
  - **Language             :*TypeScript*

### Other

https://github.com/awslabs/aws-solutions-constructs/blob/8130c54616c0422b7b141b4bc9d3086becdf7a2d/source/patterns/%40aws-solutions-constructs/core/lib/cloudfront-distribution-helper.ts#L168

This line of code is causing the `overrideProps` function to overwrite arrays instead of merging them

--- 

This is :bug: Bug Report

**Comment by hnishar** [comment]
andrewgadziksonos@ I'm able to reproduce the issue, the fix will be coming in the next release

**Comment by hnishar** [comment]
andrewgadziksonos@ Fix has been pushed out in v1.81.0
